test/common/extensions/filters/http/assertion/assertion_filter_test.cc

```diff
@@ -126,6 +119,29 @@ TEST_F(AssertionFilterTest, DataNoMatch) {
   EXPECT_EQ(Http::FilterDataStatus::StopIterationNoBuffer, filter_->decodeData(*body, true));
 }
 
+TEST_F(AssertionFilterTest, DataMissing) {
```

Comment:
We are missing tests for some of  the other combinations e.g, trailer missing in headers, trailer missing in data